libraries/default_handler.rb

```diff
@@ -34,36 +36,28 @@ def handle_uninstall
     private
 
     def configure_package(action_name)
-      # This is to cleanup old cruft from chef-server-ingredient
-      file '/etc/yum.repos.d/chef_stable_.repo' do
-        action :delete
-        only_if { ::File.exist?('/etc/yum.repos.d/chef_stable_.repo') }
-      end
-
       if new_resource.package_source
         configure_from_source_package(action_name)
-      elsif new_resource.channel == :unstable
-        configure_from_unstable_channel(action_name)
-      else
-        if use_custom_repo_recipe?
-          # Use the custom repository recipe.
-          include_recipe custom_repo_recipe
-        else
-          # Enable the required yum-repository.
-          include_recipe "yum-chef::#{new_resource.channel}"
-        end
-
+      elsif use_custom_repo_recipe?
+        # Use the custom repository recipe.
+        include_recipe custom_repo_recipe
         configure_from_repo(action_name)
+      else
+        configure_from_channel(action_name)
       end
     end
 
     def configure_from_source_package(action_name, local_path = nil)
-      rpm_package new_resource.product_name do
+      package new_resource.product_name do
         action action_name
         package_name ingredient_package_name
-        options new_resource.options
+        options package_options
         source local_path || new_resource.package_source
-
+        provider value_for_platform_family(
+          'debian'  => Chef::Provider::Package::Dpkg,
+          'rhel'    => Chef::Provider::Package::Rpm,
+          'windows' => Chef::Provider::Package::Windows
```

Comment:
Do we need to explicitly set a provider here? Doesn't chef do this for us automatically?